src/utilities/error.c

```diff
@@ -85,6 +85,66 @@ hypre_error_handler(const char *filename, HYPRE_Int line, HYPRE_Int ierr, const
    hypre__global_error = err;
 }
 
+/*--------------------------------------------------------------------------
+ *--------------------------------------------------------------------------*/
+
+void
+hypre_error_handler_rank_0(HYPRE_Int   myid,
+                           const char *filename,
+                           HYPRE_Int   line,
+                           HYPRE_Int   ierr,
+                           const char *msg)
+{
+   /* Update error code in all ranks */
+   hypre__global_error.error_flag |= ierr;
+
+   /* Print message only from rank 0 */
+   if (!myid)
+   {
+      hypre_error_handler(filename, line, ierr, msg);
+   }
+}
+
+/*--------------------------------------------------------------------------
+ *--------------------------------------------------------------------------*/
+
+void
+hypre_error_code_save(void)
+{
+   /* Store the current error code in a temporary variable */
+   hypre_error_temp_flag = hypre_error_flag;
+
+   /* Reset current error code */
+   HYPRE_ClearAllErrors();
+}
+
+/*--------------------------------------------------------------------------
+ *--------------------------------------------------------------------------*/
+
+void
+hypre_error_code_restore(void)
+{
+   /* Restore hypre's error code */
+   hypre_error_flag = hypre_error_temp_flag;
+
+   /* Reset temporary error code */
+   hypre_error_temp_flag = 0;
+}
+
+/*--------------------------------------------------------------------------
+ *--------------------------------------------------------------------------*/
+
+HYPRE_Int
+HYPRE_GetMaxGlobalError(MPI_Comm comm)
```

Comment:
What we really should do here is provide a "union" and just call it `GetGlobalError`.  Using max can miss errors in "smaller bits".

Reply:
Right! How about using MPI_BOR (bitwise OR) instead of MPI_MAX?

Reply:
Perfect.  Thanks!

Reply:
As a side question, since we are dealing with this...

Should we allow negative error codes? 

If not, then I propose changing `HYPRE_Int error_flag` to `hypre_uint error_flag` in the `hypre_Error` struct

Reply:
Bringing this up also because bitwise OR on signed data types might produce unexpected results...

Reply:
This might be something to do, but let's save it for another PR.  Functionally in hypre, we only ever have non-negative error codes, so bitwise OR shouldn't be a problem.

Reply:
Done in [80e65b4](https://github.com/hypre-space/hypre/pull/1040/commits/80e65b40b05de97b1f44871491092298b4b4f03a)